src/test/api/karate/cards/deleteCardsByEndDate.feature

```diff
@@ -0,0 +1,93 @@
+Feature: Delete Cards by end date before
+
+
+  Background:
+   #Getting token for admin and operator1 user calling getToken.feature
+    * def signIn = call read('../common/getToken.feature') { username: 'admin'}
+    * def authToken = signIn.authToken
+    * def signInAsTSO = call read('../common/getToken.feature') { username: 'operator1'}
+    * def authTokenAsTSO = signInAsTSO.authToken
+
+  Scenario: Post three cards in one request
+
+    * def card =
+"""
+[
+{
+	"publisher" : "api_test",
+	"processVersion" : "1",
+	"process"  :"api_test",
+	"processInstanceId" : "process1",
+	"state": "messageState",
+	"groupRecipients": ["Dispatcher"],
+	"severity" : "INFORMATION",
+	"startDate" : 1553186770000,
+	"endDate" : 1553186780000,
+	"summary" : {"key" : "defaultProcess.summary"},
+	"title" : {"key" : "defaultProcess.title"},
+	"data" : {"message":"a message"}
+}
+{
+	"publisher" : "api_test",
+	"processVersion" : "1",
+	"process"  :"api_test",
+	"processInstanceId" : "process2card1",
+	"state": "messageState",
+	"groupRecipients": ["Dispatcher"],
+	"severity" : "COMPLIANT",
+	"startDate" : 1553186770000,
+	"summary" : {"key" : "defaultProcess.summary"},
+	"title" : {"key" : "defaultProcess.title"},
+	"data" : {"message":"new message (card 1)"}
+},
+{
+	"publisher" : "api_test",
+	"processVersion" : "1",
+	"process"  :"api_test",
+	"processInstanceId" : "process2card2",
+	"state": "messageState",
+	"groupRecipients": ["Dispatcher"],
+	"severity" : "COMPLIANT",
+	"startDate" : 1553186770000,
+	"endDate" : 1553186880999,
+	"summary" : {"key" : "defaultProcess.summary"},
+	"title" : {"key" : "defaultProcess.title"},
+	"data" : {"message":"new message (card2) "}
+}
+]
+"""
+
+# Push 3 card
+    Given url opfabPublishCardUrl + 'cards'
+    And request card
+    When method post
+    Then status 201
+    And match response.count == 3
+
+
+  Scenario: Delete the cards before date as admin
+
+# delete all cards before date
+    Given url opfabPublishCardUrl + 'cards?' + 'endDateBefore=1553186800000'
+	And header Authorization = 'Bearer ' + authToken
+    When method delete
+    Then status 200
+
+#get deleted card should return NOT_FOUND
+    Given url opfabUrl + 'cards/cards/api_test.process1'
+    And header Authorization = 'Bearer ' + authTokenAsTSO
+    When method get
+    Then status 404
+
+#get deleted card should return NOT_FOUND
+    Given url opfabUrl + 'cards/cards/api_test.process2card1'
+    And header Authorization = 'Bearer ' + authTokenAsTSO
+    When method get
+    Then status 404
+
+#get card with user operator1 
+    Given url opfabUrl + 'cards/cards/api_test.process2card2'
+    And header Authorization = 'Bearer ' + authTokenAsTSO
+    When method get
+    Then status 200
+
```

Comment:
Delete the remaining cards to clean cards mongo collection for next tests

Reply:
Done